README.md
# DoSomething.org Platform
This is the Drupal 7 app powering [DoSomething.org](http://www.dosomething.org/) and our international affiliates. Yes, this is our production website, and yes, [it's open-source](https://blog.dosomething.org/we-open-sourced-our-code-heres-why-you-should-too/). Feel free to fork this repository, submit pull requests, or use it as a reference for your own Drupal website.

# Getting Started
Documentation for how to get started can be found in the [wiki](https://github.com/DoSomething/phoenix/wiki).

# License
&copy;2015 DoSomething.org. DoSomething is free software, and may be redistributed under the terms specified in the [LICENSE](https://github.com/DoSomething/phoenix/blob/dev/LICENSE) file. The name and logo for DoSomething.org are trademarks of Do Something, Inc and may not be used without permission.


CHANGELOG.md
# Change Log

## [v2.0.17](https://github.com/DoSomething/phoenix/tree/v2.0.17) (2015-11-16)
[Full Changelog](https://github.com/DoSomething/phoenix/compare/v2.0.16...v2.0.17)

**Fixed bugs:**

- A black rectangle lingers after removing a video on the campaign template [\#5773](https://github.com/DoSomething/phoenix/issues/5773)

**Closed issues:**

- Enable CSP \(Content-Security-Policy\) Within Phoenix [\#5392](https://github.com/DoSomething/phoenix/issues/5392)

**Merged pull requests:**

- Enables CSP \(Content-Security-Policy\) in Report-only mode [\#5774](https://github.com/DoSomething/phoenix/pull/5774) ([sergii-tkachenko](https://github.com/sergii-tkachenko))

## [v2.0.16](https://github.com/DoSomething/phoenix/tree/v2.0.16) (2015-11-13)
[Full Changelog](https://github.com/DoSomething/phoenix/compare/v2.0.15...v2.0.16)

**Fixed bugs:**

- Campaigns on SMS Game Confirmation Page go to /campaigns [\#5673](https://github.com/DoSomething/phoenix/issues/5673)

**Closed issues:**

- Increase Root Volume Size for Inode Usage on All Drupal Webheads [\#5771](https://github.com/DoSomething/phoenix/issues/5771)
- Upgrade to Node 4.0 on Prod Web Servers [\#5770](https://github.com/DoSomething/phoenix/issues/5770)
- Change translatable fields language on production [\#5724](https://github.com/DoSomething/phoenix/issues/5724)

**Merged pull requests:**

- Before calling theme on video id, check there's a value in the field. [\#5776](https://github.com/DoSomething/phoenix/pull/5776) ([angaither](https://github.com/angaither))
- Builds a correct global url when getting campaign block vars [\#5772](https://github.com/DoSomething/phoenix/pull/5772) ([deadlybutter](https://github.com/deadlybutter))

## [v2.0.15](https://github.com/DoSomething/phoenix/tree/v2.0.15) (2015-11-12)
[Full Changelog](https://github.com/DoSomething/phoenix/compare/v2.0.14...v2.0.15)

**Merged pull requests:**

- Update grunt-sass version. [\#5769](https://github.com/DoSomething/phoenix/pull/5769) ([DFurnes](https://github.com/DFurnes))

## [v2.0.14](https://github.com/DoSomething/phoenix/tree/v2.0.14) (2015-11-12)
[Full Changelog](https://github.com/DoSomething/phoenix/compare/v2.0.13...v2.0.14)

**Merged pull requests:**

- Fixes Array showing on static content [\#5768](https://github.com/DoSomething/phoenix/pull/5768) ([deadlybutter](https://github.com/deadlybutter))

## [v2.0.13](https://github.com/DoSomething/phoenix/tree/v2.0.13) (2015-11-12)
[Full Changelog](https://github.com/DoSomething/phoenix/compare/v2.0.12...v2.0.13)

**Fixed bugs:**

- /br/sobre/faqs and /mx/sobre/contacto throw error messages [\#5747](https://github.com/DoSomething/phoenix/issues/5747)
- Cause filter of finder should be in two columns on thor [\#5682](https://github.com/DoSomething/phoenix/issues/5682)

**Closed issues:**

- Disable X-Frame-Options header [\#5742](https://github.com/DoSomething/phoenix/issues/5742)
- Recommended Campaign fields to rename [\#4506](https://github.com/DoSomething/phoenix/issues/4506)

**Merged pull requests:**

- Disables x frame options [\#5767](https://github.com/DoSomething/phoenix/pull/5767) ([sergii-tkachenko](https://github.com/sergii-tkachenko))
- Removes entity usage in static content long text preprocess [\#5766](https://github.com/DoSomething/phoenix/pull/5766) ([deadlybutter](https://github.com/deadlybutter))

## [v2.0.12](https://github.com/DoSomething/phoenix/tree/v2.0.12) (2015-11-11)
[Full Changelog](https://github.com/DoSomething/phoenix/compare/v2.0.11...v2.0.12)

**Closed issues:**

- Do not return last\_name field in user object when using `load\_user=true` [\#5756](https://github.com/DoSomething/phoenix/issues/5756)

**Merged pull requests:**

- Adding another term and fixing how it is wrapped in t\(\). [\#5765](https://github.com/DoSomething/phoenix/pull/5765) ([weerd](https://github.com/weerd))
- Missing translations 07 olé [\#5761](https://github.com/DoSomething/phoenix/pull/5761) ([weerd](https://github.com/weerd))
- removes last\_name in returned user [\#5757](https://github.com/DoSomething/phoenix/pull/5757) ([chloealee](https://github.com/chloealee))

## [v2.0.11](https://github.com/DoSomething/phoenix/tree/v2.0.11) (2015-11-11)
[Full Changelog](https://github.com/DoSomething/phoenix/compare/v2.0.10...v2.0.11)

**Fixed bugs:**

- field\_created\_copy content got wiped on production [\#5735](https://github.com/DoSomething/phoenix/issues/5735)
- 404 Not Fount nginx/1.4.6 \(Ubuntu\) error when trying to access MX Campaign [\#5667](https://github.com/DoSomething/phoenix/issues/5667)
- Translated content is shown as cached arrays [\#5302](https://github.com/DoSomething/phoenix/issues/5302)
- Inorrect Campaign Links on SMS Game Confirmation Page [\#4836](https://github.com/DoSomething/phoenix/issues/4836)

**Closed issues:**

- Privacy Policy and Terms of Service 404 for BR/MX users [\#5745](https://github.com/DoSomething/phoenix/issues/5745)
- `/reportbacks.json?load\_user=true` and http://dev.dosomething.org:8888/api/v1/reportback-items.json?load\_user=true broken on staging   [\#5741](https://github.com/DoSomething/phoenix/issues/5741)
- Global Transactional Email Updates [\#5630](https://github.com/DoSomething/phoenix/issues/5630)
- Yahoo: Remove yahoo partnership from fact pages [\#5406](https://github.com/DoSomething/phoenix/issues/5406)
- Migrate to MariaDB RDS from MySQL RDS [\#5224](https://github.com/DoSomething/phoenix/issues/5224)

**Merged pull requests:**

- Change additional text type from text, to formatted text. [\#5760](https://github.com/DoSomething/phoenix/pull/5760) ([angaither](https://github.com/angaither))
- Adding missing term to POT file. [\#5755](https://github.com/DoSomething/phoenix/pull/5755) ([weerd](https://github.com/weerd))
- Revising a term and updating styles with how it is rendered. [\#5749](https://github.com/DoSomething/phoenix/pull/5749) ([weerd](https://github.com/weerd))

config/settings.php
<?php

/**
 * Lets do something about globals
 */
define('DS_PROFILE_PATH', 'profiles/dosomething');
define('DS_MODULES_PATH', DS_PROFILE_PATH . '/modules');
define('DS_THEMES_PATH', DS_PROFILE_PATH . 'themes');
define('DS_LIBRARIES_PATH', DS_PROFILE_PATH . '/libraries');

/**
 * Database settings:
 */
$databases['default']['default'] = array(
  'database' => getenv('DS_DB_MASTER_NAME') ?: 'dosomething',
  'username' => getenv('DS_DB_MASTER_USER') ?: 'root',
  'password' => getenv('DS_DB_MASTER_PASS') ?: '',
  'host' => getenv('DS_DB_MASTER_HOST') ?: 'localhost',
  'port' => getenv('DS_DB_MASTER_PORT') ?: '3306',
  'driver' => getenv('DS_DB_MASTER_DRIVER') ?: 'mysql',
  'prefix' => getenv('DS_DB_MASTER_PREFIX') ?: '',
);

/**
 * Hosts & urls
 */
$hostname = $_SERVER['HTTP_HOST'] ?: 'dev.dosomething.org';

$protocol = (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off' || $_SERVER['SERVER_PORT'] == 443) ? "https://" : "http://";

$insecure_port = getenv('DS_INSECURE_PORT') ?: 8888;
$secure_port = getenv('DS_SECURE_PORT') ?: 8889;

// $hostname may already have port included. Check it before you wreck it!
if (!preg_match('/.+:[0-9]+$/', $hostname) && !empty($insecure_port)
    && ($insecure_port != 80)) {

  $base_url = $protocol . $hostname . ':' . $insecure_port;
}
else {
  $base_url = $protocol . $hostname;
}

$conf['https'] = TRUE;

/**
 * Caching
 */
// Add Varnish as the page cache handler.
$conf['varnish_version'] = '3';
$conf['cache_backends'] = array('profiles/dosomething/modules/contrib/varnish/varnish.cache.inc');
$conf['cache_class_cache_page'] = 'VarnishCache';

// This is managed from salt://varnishd/secret
$conf['varnish_control_key'] = '00c9203c65874ca5b4c359e19f00bf56';

// Drupal 7 does not cache pages when we invoke hooks during bootstrap.
// This needs to be disabled.
$conf['page_cache_invoke_hooks'] = FALSE;

// Setup Redis cache backend using Predis PHP library.
$conf['redis_client_interface'] = 'Predis';

// Register Predis classes autoload.
// @see Redis_Client_Predis::setPredisAutoload().
define('PREDIS_BASE_PATH', DRUPAL_ROOT . '/' . DS_LIBRARIES_PATH . '/predis/lib/');
require_once DS_MODULES_PATH . '/contrib/redis/redis.autoload.inc';
$conf['cache_backends'][] = DS_MODULES_PATH . '/contrib/redis/redis.autoload.inc';

// Redis client settings.
$conf['redis_client_host'] = getenv('DS_REDIS_HOST') ?: '127.0.0.1';
$conf['redis_client_port'] = getenv('DS_REDIS_PORT') ?: '6379';

// Enforce a common cache key.
$conf['cache_prefix'] = 'ds';

// Cache everything in Redis, except the form cache.
$conf['cache_default_class'] = 'Redis_Cache';
$conf['cache_class_cache_form'] = 'DrupalDatabaseCache';

$conf['lock_inc'] = DS_MODULES_PATH . '/contrib/redis/redis.lock.inc';

/**
 * Salt hash:
 */
$drupal_hash_salt = '3i_SZ1VTl_8FBxXeZhTEvf6LkeVNypM0EV90tNuHs5k';

/**
 * PHP Configuration overrides:
 */
ini_set('session.gc_probability', 1);
ini_set('session.gc_divisor', 100);
ini_set('session.gc_maxlifetime', 200000);
ini_set('session.cookie_lifetime', 2000000);

/**
 * Fast 404 pages:
 */
$conf['404_fast_paths_exclude'] = '/\/(?:styles)\//';
$conf['404_fast_paths'] = '/\.(?:txt|png|gif|jpe?g|css|js|ico|swf|flv|cgi|bat|pl|dll|exe|asp)$/i';
$conf['404_fast_html'] = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.0//EN" "http://www.w3.org/MarkUp/DTD/xhtml-rdfa-1.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>404 Not Found</title></head><body><h1>Not Found</h1><p>The requested URL "@path" was not found on this server.</p></body></html>';

/**
 * Load environment aware settings override files
 */
$environment = getenv('DS_ENVIRONMENT') ?: 'local';

// Include local settings file if it exists.
if (is_readable('sites/default/settings.'. $environment .'.php')) {
  include_once('sites/default/settings.'. $environment .'.php');
}

$conf['optimizely_id'] = '747623297';

$conf['dosomething_is_affiliate'] = FALSE;

// The 'solr' hostname must be defined in /etc/hosts.
$conf['apachesolr_host'] = getenv('DS_APACHESOLR_HOST') ?: 'solr';
$conf['apachesolr_port'] = getenv('DS_APACHESOLR_PORT') ?: '8080';

$conf_path = explode('/', conf_path());
$solr_path = $conf_path[1] == 'default' ? 'collection1' : $conf_path[1];
$conf['apachesolr_path'] = "solr/{$solr_path}";

// This is different from the apachesolr host because it's on the client side
$conf['dosomething_search_finder_url'] = (getenv('DS_FINDER_URL') ?: '//search.dosomething.org') . "/solr/";

$conf['dosomething_search_finder_collection'] = $solr_path;
